peachjam/models/external_document.py

```diff
@@ -0,0 +1,21 @@
+from django.utils import timezone
+from django.utils.text import slugify
+from languages_plus.models import Language
+
+from .core_document_model import CoreDocument
+
+
+class ExternalDocument(CoreDocument):
+
+    frbr_uri_doctypes = ["doc"]
+
+    def clean(self):
+        self.doc_type = "external"
+        self.frbr_uri_number = slugify(self.title)
+        self.frbr_uri_doctype = "doc"
```

Comment:
This is already enforced in the model and (I think) the admin because `frbr_uri_doctypes` is set on the model.

Reply:
I'm setting these values here on the clean method rather than the save method to allow the  validations to pass,

Reply:
If you pre-populate the form, then the validation will pass automatically. That's the best place to provide defaults, I think. Doing them here means the user can't override them at all.

Reply:
Changed